lib/turn.rb

```diff
@@ -0,0 +1,59 @@
+class Turn
+  attr_reader :player1, :player2, :spoils_of_war
+
+  def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2
+    @spoils_of_war = []
+  end
+
+  def type
+    if player1.deck.rank_of_card_at(0) != player2.deck.rank_of_card_at(0)
+      :basic
+    elsif player1.deck.rank_of_card_at(0) == player2.deck.rank_of_card_at(0) &&
+      player1.deck.rank_of_card_at(2) != player2.deck.rank_of_card_at(2)
+      :war
+    else
+      :mutually_assured_destruction
+    end
+  end
+
+  def winner
+    if type == :basic
+      if player1.deck.rank_of_card_at(0) > player2.deck.rank_of_card_at(0)
+        then player1
+      else player2
+      end
+
+    elsif type == :war
+      if player1.deck.rank_of_card_at(2) > player2.deck.rank_of_card_at(2)
+        then player1
+      else player2
+      end
+
+    else type == :mutually_assured_destruction
+      "No Winner"
+    end
+  end
```

Comment:
Overall this method is good! It works! A couple ruby style convention things I noticed to keep an eye out for:
* we don't need `then` in our `if` statements
* We should use the instance variable in the class where its defined rather than the method (so `@player1` instead of `player1`)
* We should put the action for the `else` outcome on a new line
* No need for an extra line break between our `if`, `elsif`, and `else`
Revised to follow those conventions would look something like this:

```ruby
  def winner
    if type == :basic
      if @player1.deck.rank_of_card_at(0) > @player2.deck.rank_of_card_at(0)
        @player1
      else
        @player2
      end
    elsif type == :war
      if @player1.deck.rank_of_card_at(2) > @player2.deck.rank_of_card_at(2)
        @player1
      else
        @player2
      end
    else type == :mutually_assured_destruction
      "No Winner"
    end
  end
```